docs: Update 12-factor how-to landing page (#2326)
canonical/charmcraft: docs/howto/manage-web-app-charms/index.rst

These guides walk you through all ways you can manage 12-factor app charms,
from initialization to usage.

:ref:`Extensions <profile>` add customized elements to a charm for specific
web app frameworks. While the overall basic logic is the same as

> :ref:`managing a charm <manage-charms>`, the following guides are relevant
> to the 12-factor app charm workflow.

Reviewer: Consider:

```suggestion
:ref:`managing a charm <manage-charms>`, the following guides are about
the 12-factor app charm workflow.
```

Not a big fan of "relevant to".

Initialization
--------------

You need both a rock and a charm to have a deployable 12-factor app in Juju.

The charm initialization step uses the ``--profile`` flag to cater the initial

> charm files for your specific web app framework.

Reviewer: Consider:

```suggestion
charm files for your specific framework.
```
